### PR TITLE
BZ2025941: Targeted must-gather CRs and logs

### PR DESCRIPTION
**Pull request by apinnick**
https://bugzilla.redhat.com/show_bug.cgi?id=2025941

Preview: https://deploy-preview-256--forklift-documentation.netlify.app/documentation/doc-migration_toolkit_for_virtualization/master/#collected-logs-cr-info_forklift

### CODE REVIEW

**Comment by apinnick** [comment]
@mguetta1  Ready for QE review.

**Comment by aufi** [comment]
Looks good, just note on gathered objects list:
image

More tech.details could be found in https://github.com/konveyor/forklift-must-gather/pull/36 (preview https://github.com/aufi/forklift-must-gather/tree/targeted-docs)

**Review by mguetta1 — CHANGES_REQUESTED** [review]

File: documentation/modules/collected-logs-cr-info.adoc [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+You can download logs and custom resource (CR) `yaml` files for the following targets by using the {project-short} web console or command line interface (CLI):
+
+* Migration plan: Web console or CLI.
+* Virtual machine: Web console or CLI.
+* Namespace: CLI only.
+
+The `must-gather` tool collects the following logs and CR files in an archive file:
+
+* Logs:
+** `importer` pod: Data volume conversion log.
```
Comment (apinnick) [inline documentation/modules/collected-logs-cr-info.adoc:17]: @istein1 Is this OK as a description of the importer pod?

File: documentation/modules/collected-logs-cr-info.adoc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+You can download logs and custom resource (CR) `yaml` files for the following targets by using the {project-short} web console or the command line interface (CLI):
+
+* Migration plan: Web console or CLI.
+* Virtual machine: Web console or CLI.
+* Namespace: CLI only.
++
+[NOTE]
+====
+If you specify a non-existent target, the `must-gather` archive file is empty.
+
```
Comment (mguetta1) [inline documentation/modules/collected-logs-cr-info.adoc:17]: No need for this note here. It is relevant only for CLI and the note in CLI section is good

File: documentation/modules/collected-logs-cr-info.adoc [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+The `must-gather` tool collects the following logs and CR files in an archive file:
+
+* Logs:
+** `importer` pod: Data volume conversion log.
+** `conversion` pod: VM conversion log. The `conversion` pod runs `virt-v2v`, which installs and configures device drivers on the PVCs of the VM.
+** `virt-launcher` pod: VM launcher log. When a migrated VM is powered on, the `virt-launcher` pod runs `QEMU-KVM` with the PVCs attached as VM disks.
+** `forklift-controller` pod: Filtered for the migration plan, virtual machine, or namespace.
+** `forklift-must-gather-api` pod: Filtered for the migration plan, virtual machine, or namespace. This log is not available if you run a targeted `must-gather` command from the CLI.
+
```
Comment (mguetta1) [inline documentation/modules/collected-logs-cr-info.adoc:29]: Please remove this line: "This log is not available if you run a targeted `must-gather` command from the CLI."
It is not true for all cases
Thanks

**Review by mguetta1 — CHANGES_REQUESTED** [review]

File: documentation/modules/collected-logs-cr-info.adoc (hunk 1 above)
Comment (istein1) [reply documentation/modules/collected-logs-cr-info.adoc:24]: Importer pod; Disk transfer into data volume log

Comment (apinnick) [reply documentation/modules/collected-logs-cr-info.adoc:24]: That is what I meant. But I can make it clearer.

File: documentation/modules/collected-logs-cr-info.adoc [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        ├── crs
+        │   └── plan
+        │       └── mig-plan-cold.yaml
+        └── logs
+            ├── forklift-controller-67656d574-w74md
+            │   └── current.log
+            └── forklift-must-gather-api-89fc7f4b6-hlwb6
+                └── current.log
+----
+<1> `importer` pod log. The naming convention is `importer-<migration_plan>-vm-<vm_id>-<number>`.
```
Comment (apinnick) [inline documentation/modules/collected-logs-cr-info.adoc:67]: @aufi @mguetta1 

Marek, Maayan says that the RHV importer and conversion pods have names that seem to include a time stamp: 

importer: importer-mtv-api-tests-21-05-12-15-37-10-035-plan-ed90dfc6-9a17-4a8btnfh
conversion: mtv-api-tests-21-05-12-15-37-10-035-plan-ed90dfc6-9a17-4a85vcg6

Is there a reason for this or is it a bug? 

If there is no reason for it, then I think it is a bug and I would add a note in the Known Issues that RHV pods have different names.

If there is a reason for it, I will update the User Guide.

Comment (apinnick) [reply documentation/modules/collected-logs-cr-info.adoc:67]: Thanks, @aufi . In that case, I will leave it for now. If it turns out to be a bug, we can deal with it after the release. Pod naming conventions are an edge case and not going to affect the user experience, given that the must-gather archive structure is clear.

File: documentation/modules/collected-logs-cr-info.adoc [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        ├── crs
+        │   └── plan
+        │       └── mig-plan-cold.yaml
+        └── logs
+            ├── forklift-controller-67656d574-w74md
+            │   └── current.log
+            └── forklift-must-gather-api-89fc7f4b6-hlwb6
+                └── current.log
+----
+<1> `importer` pod log. The naming convention is `importer-<migration_plan>-vm-<vm_id>-<number>` for VMware and `importer-<migration_plan>-<partial_vm_id>` for {rhv-full}.
```
Comment (mguetta1) [inline documentation/modules/collected-logs-cr-info.adoc:67]: The naming convention is `importer-<migration_plan>-vm-<vm_id>-<random_string/5 letters>` for VMware and `importer-<migration_plan>-<partial_vm_id><random_string/5 letters>` for {rhv-full}
Example for RHV:
importer-mtv-api-tests-21-05-12-15-37-10-035-plan-ed90dfc6-9a17-4a8btnfh
Plan name: mtv-api-tests-21-05-12-15-37-10-035-plan
RHV ID **beginning**: ed90dfc6-9a17-4a8
5 more letters: btnfh

File: documentation/modules/collected-logs-cr-info.adoc [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+<3> `conversion` pod log. The naming convention is `<migration_plan>-vm-<vm_id>-<number>`for VMware and `<migration_plan>-<partial_vm_id>` for {rhv-full}.
```
Comment (mguetta1) [inline documentation/modules/collected-logs-cr-info.adoc:68]: The naming convention is `<migration_plan>-vm-<vm_id>-<random_string/5 letters>` for VMware and `<migration_plan>-<partial_vm_id><random_string/5 letters>` for {rhv-full}.
Example for RHV:
mtv-api-tests-21-05-12-15-37-10-035-plan-ed90dfc6-9a17-4a85vcg6
Plan name: mtv-api-tests-21-05-12-15-37-10-035-plan
RHV ID **beginning**: ed90dfc6-9a17-4a8
5 more letters: 5vcg6

**Review by mguetta1 — CHANGES_REQUESTED** [review]

File: documentation/modules/collected-logs-cr-info.adoc [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        ├── crs
+        │   └── plan
+        │       └── mig-plan-cold.yaml
+        └── logs
+            ├── forklift-controller-67656d574-w74md
+            │   └── current.log
+            └── forklift-must-gather-api-89fc7f4b6-hlwb6
+                └── current.log
+----
+<1> `importer` pod log. The naming convention is `importer-<migration_plan>-vm-<vm_id>-<5_char_id>` for VMware and `importer-<migration_plan>-<partial_vm_id>-<5_char_id>` for {rhv-full}.
```
Comment (mguetta1) [inline documentation/modules/collected-logs-cr-info.adoc:67]: <partial_vm_id><5_char_id>
No hyphen here

File: documentation/modules/collected-logs-cr-info.adoc [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+<3> `conversion` pod log. The naming convention is `<migration_plan>-vm-<vm_id>-<5_char_id>`for VMware and `<migration_plan>-<partial_vm_id>-<5_char_id>` for {rhv-full}.
```
Comment (mguetta1) [inline documentation/modules/collected-logs-cr-info.adoc:68]: 1. `for VMware - with space is missing " for VMware"
2. <partial_vm_id><5_char_id> - No hyphen here

**Review by mguetta1 — CHANGES_REQUESTED** [review]

File: documentation/modules/collected-logs-cr-info.adoc [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        ├── crs
+        │   └── plan
+        │       └── mig-plan-cold.yaml
+        └── logs
+            ├── forklift-controller-67656d574-w74md
+            │   └── current.log
+            └── forklift-must-gather-api-89fc7f4b6-hlwb6
+                └── current.log
+----
+<1> `importer` pod log. The naming convention is `importer-<migration_plan>-vm-<vm_id>-<5_char_id>` for VMware and `importer-<migration_plan>-<partial_vm_id>-<5_char_id>` for {rhv-full}.
```
Comment (apinnick) [inline documentation/modules/collected-logs-cr-info.adoc:67]: @mguetta1 
Please confirm: For VMware, is "vm" part of the "VM ID"? If so, I need to remove `-vm-` from the string for the VMware naming convention.

Comment (mguetta1) [reply documentation/modules/collected-logs-cr-info.adoc:67]: @apinnick I think yes. I didn't realize that before

**Review by mguetta1 — APPROVED** [review]
Looks good to me!

**Comment by aufi** [comment]
:+1:

**Pull request merged**